src/scripts/modules/search/advanced/advanced.coffee

```diff
@@ -69,11 +68,14 @@ define (require) ->
       query = @formatQuery(values)
 
       @search(query)
-      if !query
-        $("#alert-danger").show()
 
     search: (query) ->
-      router.navigate("search?q=#{query}", {trigger: true})
+      if !query
+        $("#alert-danger").show()
+      else if query == 'sort:popularity' || query == 'sort:pubDate'
+          $("#alert-danger").show()
+      else if query
+        router.navigate("search?q=#{query}", {trigger: true})
```

Comment:
It might yield more stable behavior to run this validation on `values` rather than `query`, to avoid string comparison

Reply:
@tomjw64 updated it! Can you re-review?